content/blog/peace-of-mind-with-cloud-secret-providers/index.md

```diff
@@ -0,0 +1,669 @@
+config:
+  aws:region: us-west-2
+```
+
+## Use your Encrypted Secret
+
+Now we've initialized our stack we can add a secret configuration value using the secret flag and see the encrypted config value in the stack configuration:
+
+```bash
+pulumi config set --secret <projectname>:supersecret correct-horse-battery-stable
+```
+
+## Verify the encryption
+
+To verify that the secret is indeed only accessible to the KMS key we created earlier, we can remove access to the key temporarily and try to perform a Pulumi operation. First, let's use the secret in our stack:
+
+{{< chooser language "javascript,typescript,python,go,csharp" >}}
+
+{{% choosable language typescript %}}
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+
+const config = new pulumi.Config();
+export const superSecret = config.require("supersecret");
+```
+
+{{% /choosable %}}
+
+{{% choosable language javascript %}}
+
+```javascript
+"use strict";
+const pulumi = require("@pulumi/pulumi");
+
+const config = new pulumi.Config();
+exports.superSecret = config.require("supersecret");
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+import pulumi
+config = pulumi.Config()
+
+pulumi.export('superSecret',  config.require("supersecret"))
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```go
+package main
+
+import (
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi/config"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		config := config.New(ctx, "")
+		superSecret := config.Require("supersecret")
+
+		ctx.Export("superSecret", pulumi.String(superSecret))
+		return nil
+	})
+}
+```
+
+{{% /choosable %}}
+
+{{% choosable language csharp %}}
+
+```csharp
+using Pulumi;
+
+class AnotherStack : Stack
+{
+    public AnotherStack()
+    {
+        var config = new Config();
+        this.SuperSecret = Output.Create(config.Require("supersecret"));
+    }
+
+    [Output("superSecret")] public Output<string> SuperSecret { get; set; }
+}
+```
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+Now we need to verify if the value is _actually_ encrypted. An easy way to do that is to try and export the secret value without access to the key. Let's unset the AWS_PROFILE environment variable and then rerun `Pulumi up`:
+
+```bash
+unset AWS_PROFILE
+pulumi up
+error: getting secrets manager: secrets (code=Unknown): InvalidSignatureException: The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.
+	status code: 400, request id: 9cf7508d-a7d5-40bb-b40f-98f68e82ac74
+```
+
+Excellent! We can't read these values without access to this KMS key. We can be safe in the knowledge our secret values are only readable by us, including when stored in the Pulumi state!
```

Comment:
The one last thing to potentially highlight is that these secrets are secret not just in the config file, but *also* in the checkpoint.  Create a resource using the secret as an input, and then `pulumi stack export` and show that the values for that resource are stored encrypted in the state file.